modules/ballerina-core/src/main/java/org/ballerinalang/model/util/JsonNode.java

```diff
@@ -0,0 +1,398 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.model.util;
+
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.ByteArrayOutputStream;
```

Comment:
Can we move IO related exception usages from the JsonNode? I am under the impression that JsonNode is just a bean class. Correct me if I am wrong.

Reply:
This is there because we do some IO related operations in toString to serialize the current node to a  String, which is done as a convenience method, which in turns calls the JsonGenerator related operations.

Reply:
Noted.